src/menu/views/file_info.c

```diff
@@ -202,111 +236,139 @@ static void process (menu_t *menu) {
+    text_y += fragment_textf(text_x, text_y, "  Attributes:\n\n  %s%s%s%s%s\n",
+                ((info.fattrib & AM_DIR) ? "  Directory" : "File"),
+                ((info.fattrib & AM_RDO) ? " | ReadOnly" : ""),
+                ((info.fattrib & AM_SYS) ? " | System" : ""),
+                ((info.fattrib & AM_ARC) ? " | Archive" : ""),
+                ((info.fattrib & AM_HID) ? " | Hidden" : "")
+                );
+
+    text_y += fragment_textf(text_x, text_y, "  Modified Timestamp:\n\n  %u-%02u-%02u, %02u:%02u\n", 
+                (info.fdate >> 9) + 1980,
+                info.fdate >> 5 & 15,
+                info.fdate & 31,
+                info.ftime >> 11,
+                info.ftime >> 5 & 63
+                );
+
+    text_y += fragment_textf(text_x, text_y, "  Type:\n\n  %s\n", format_file_type());
+}
+
+static void menu_fileinfo_draw_n64_rom_info(surface_t *d, layout_t *layout) {
+
+    const int text_x = layout->offset_x + layout->offset_text_x;
+    int text_y = layout->offset_y + layout->offset_text_y;
+    text_y += fragment_textf(text_x, text_y, "File Information:\n\n");
+    text_y += fragment_textf(text_x, text_y, "  Name:\n\n  %s\n", info.fname);
+    text_y += fragment_textf(text_x, text_y, "  Size:\n\n  %d Bytes\n", (int)info.fsize);
+    text_y += fragment_textf(text_x, text_y, "  Modified Timestamp:\n\n  %u-%02u-%02u, %02u:%02u\n", 
+            (info.fdate >> 9) + 1980,
+            info.fdate >> 5 & 15,
+            info.fdate & 31,
+            info.ftime >> 11,
+            info.ftime >> 5 & 63
+            );
+
+    text_y += fragment_textf(text_x, text_y, "\n");
+    text_y += fragment_textf(text_x, text_y, "N64 ROM Information:\n\n");
+    text_y += fragment_textf(text_x, text_y, "  Endian: %s\n", format_rom_endian(rom_header.endian));
+    text_y += fragment_textf(text_x, text_y, "  Title: %s\n", rom_header.title);
+    text_y += fragment_textf(text_x, text_y, "  Media Type: %c - %s\n", rom_header.metadata.media_type, format_rom_media_type(rom_header.metadata.media_type));
+    text_y += fragment_textf(text_x, text_y, "  Unique ID: %.2s\n", (char*)&(rom_header.metadata.unique_identifier));
+    text_y += fragment_textf(text_x, text_y, "  Destination Market: %c - %s\n", rom_header.metadata.destination_market, format_rom_destination_market(rom_header.metadata.destination_market));
+    text_y += fragment_textf(text_x, text_y, "  Version: %hhu\n", rom_header.version);
+    text_y += fragment_textf(text_x, text_y, "  Checksum: 0x%016llX\n", rom_header.checksum);
+    text_y += fragment_textf(text_x, text_y, "  Save Type: %s\n", format_rom_save_type(rom_db_match_save_type(rom_header)));
+    text_y += fragment_textf(text_x, text_y, "  Expansion PAK: %s\n\n", format_rom_memory_type(rom_db_match_expansion_pak(rom_header)));
+
+    if (boxart_image != NULL && cached_boxart_image_dl != NULL)
+    {
+        rspq_block_run(cached_boxart_image_dl);
+    }
```

Comment:
Either move this after drawing actions bar (line 329) or call `fragment_text_start(text_color);` again before drawing actions bar text (line 327). Drawing sprite in the middle of text routines changes RDP mode and loaded texture and messes up font drawing.

Reply:
Thanks. That works.